src/snowflake/snowpark/async_job.py

```diff
@@ -405,8 +407,12 @@ def result(
             else:
                 raise ValueError(f"{async_result_type} is not supported")
         for action in self._post_actions:
+            query = action.sql
+            if self._placeholders:
```

Comment:
why do we need a special handling for async job ?

Reply:
because when dropping temp objects after async job is done, we need to drop the correct temp object. This information needs to be propagated to async job as well.